### ISSUE TITLE
CORS Error with Custom Headers

### ISSUE THREAD

**Issue by fenriskiba** [checkr/flagr#387]
We tried implementing the functionality for PR #300, but ran into a CORS Error when we tried to call it. It looks like Flagr is not including some of the required CORS Headers in the preflight call responses because the User ID Header being a new Custom header:

<img width="1039" alt="cors-preflight" src="https://user-images.githubusercontent.com/1434614/90154881-a7c83d80-dd58-11ea-888d-af911d995cf2.png">

I think it is an issue with the CORS Configs getting hardcoded in [middleware.go](https://github.com/checkr/flagr/blob/master/pkg/config/middleware.go#L77) with a set list of Allowed Headers. If so, it should be fairly easy to convert that (and maybe a couple other CORS Configs) to be populated by a Env Config, but I wanted to check if that made sense as the problem before starting the process on our side to contribute that change.

**Comment by marceloboeira** [comment]
I've always ran flagr under a Nginx instance and I would solve such issues on that level, but it seems a fair usage. @zhouzhuojie any thoughts?

**Comment by fenriskiba** [comment]
*facepalm*
We encountered this in a local environment with a local instance of Flagr. We do have a proxy layer we can handle it with that I forgot to take into account.

I'd definitely still like to get this fixed so we don't have to deal with running the proxy in that local environment though, but that does make this a lot lower priority for us.

**Comment by zhouzhuojie** [comment]
Yep, makes sense. I have a PR to config CORS
